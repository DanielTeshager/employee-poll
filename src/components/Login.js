import React, { useState } from "react";
import { connect } from "react-redux";
import { login } from "../redux/auth";
import "@testing-library/jest-dom/extend-expect";

const Login = ({ users, handleLogin }) => {
	const [selectedUser, setSelectedUser] = useState("");

	const handleUserSelect = (event) => {
		setSelectedUser(event.target.value);
	};

	const handleLoginSubmit = (event) => {
		event.preventDefault();
		if (selectedUser !== "") {
			handleLogin(selectedUser);
		}
	};

	return (
		<div>
			<h2>Login</h2>
			<form onSubmit={handleLoginSubmit}>
				{/* add label for users */}
				<label htmlFor="users">Select User</label>
				<br></br>
				<select value={selectedUser} onChange={handleUserSelect} id="users">
					<option value="">Select User</option>
					{Object.keys(users).map((userId) => (
						<option key={userId} value={userId}>
							{users[userId].name}
						</option>
					))}
				</select>
				<button type="submit">Login</button>
			</form>
		</div>
	);
};

const mapStateToProps = (state) => ({
	users: state.auth.users,
});

const mapDispatchToProps = (dispatch) => ({
	handleLogin: (userId) => dispatch(login(userId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Login);
