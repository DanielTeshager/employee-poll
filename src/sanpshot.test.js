import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import Login from "./components/Login";

const mockStore = configureStore();

describe("Login", () => {
	it("renders correctly", () => {
		const initialState = {
			auth: {
				users: {
					sarahedo: {
						id: "sarahedo",
						name: "Sarah Edo",
						avatarURL: "https://tylermcginnis.com/would-you-rather/sarah.jpg",
						answers: {},
						questions: [],
					},
				},
			},
		};
		const store = mockStore(initialState);

		const tree = renderer
			.create(
				<Provider store={store}>
					<Login />
				</Provider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
