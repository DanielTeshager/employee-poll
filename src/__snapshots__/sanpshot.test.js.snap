// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div>
  <h2>
    Login
  </h2>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="users"
    >
      Select User
    </label>
    <br />
    <select
      id="users"
      onChange={[Function]}
      value=""
    >
      <option
        value=""
      >
        Select User
      </option>
      <option
        value="sarahedo"
      >
        Sarah Edo
      </option>
    </select>
    <button
      type="submit"
    >
      Login
    </button>
  </form>
</div>
`;
